folium/folium.py

```diff
@@ -951,44 +951,37 @@ def json_style(style_cnt, line_color, line_weight, line_opacity,
         self.template_vars.setdefault('gjson_layers', []).append(layer)
 
     @iter_obj('image_overlay')
-    def image_overlay(self, data, opacity=0.75, min_lat=-90.0, max_lat=90.0,
+    def image_overlay(self, data, image_opacity=1, min_lat=-90.0, max_lat=90.0,
                       min_lon=-180.0, max_lon=180.0, image_name=None,
-                      filename=None, data_projection='mercator'):
-        """Simple image overlay of raster data from a numpy array.  This is a
-        lightweight way to overlay geospatial data on top of a map.
-        If your data is high res, consider implementing a WMS server
-        and adding a WMS layer.
-
-        This function works by generating a PNG file from a numpy
-        array.  If you do not specify a filename, it will embed the
-        image inline.  Otherwise, it saves the file in the current
-        directory, and then adds it as an image overlay layer in
-        leaflet.js.  By default, the image is placed and stretched
-        using bounds that cover the entire globe.  By default, we
-        assume that your data is in geodetic projection and thus
-        project it to web mercator for display purposes.  If you are
-        overlaying a non-georeferenced image, set data_projection to
-        None.
+                      filename=None):
+        """
+        Simple image overlay of raster `data` from an image or a NumPy array.
+        This is a lightweight way to overlay geospatial data on top of a map.
+        The default lon, lat corners are the whole globe.  Adjust it to your
+        image.
+
+        If `data` is is a NumPy array a PNG file will be created for you.
+        If you do not specify a filename the image will be embedded inline.
+
+        NOTE: The image must be project as Web Mercator
+        (https://en.wikipedia.org/wiki/Web_Mercator).  See the examples below
+        for re-projection.
 
         Parameters
         ----------
-        data: numpy array OR url string, required.
-            if numpy array, must be a image format,
-            i.e., NxM (mono), NxMx3 (RGB), or NxMx4 (RGBA)
-            if url, must be a valid url to a image (local or external)
-        opacity: float, default 0.75
-            Image layer opacity in range 0 (transparent) to 1 (opaque)
-        min_lat: float, default -90.0
-        max_lat: float, default  90.0
-        min_lon: float, default -180.0
-        max_lon: float, default  180.0
+        data: NumPy array OR image path.
```

Comment:
Virtually all the external data file (GeoJSONs, images) can be local or a URL.  To keep it consistent with the other docstrings I will leave just `image` ~~path~~.